<h1><center>Ecommerce Segmentation Analysis<center/><h1/>
<h2>April 2025<h2/>
Celine Ng


Table of Contents
1. Notebook Preparation
    1. Libraries Installation
    1. Data loading
2. Data Cleaning
3. EDA

# 1. Notebook Preparation

## 1.1. Libraries Installation

In [8]:
from utils.eda import *

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 1.2. Data loading

In [4]:
data = pd.read_csv('data/data.csv')
data.head()

Customer_ID  Age  Gender Income_Level Marital_Status Education_Level  \
0  37-611-6911   22  Female       Middle        Married      Bachelor's   
1  29-392-9296   49    Male         High        Married     High School   
2  84-649-5117   24  Female       Middle         Single        Master's   
3  48-980-6078   29  Female       Middle         Single        Master's   
4  91-170-9072   33  Female       Middle        Widowed     High School   

  Occupation  Location     Purchase_Category Purchase_Amount  ...  \
0     Middle      Évry  Gardening & Outdoors        $333.80   ...   
1       High  Huocheng      Food & Beverages        $222.22   ...   
2       High    Huzhen       Office Supplies        $426.22   ...   
3     Middle    Wiwilí       Home Appliances        $101.31   ...   
4     Middle      Nara             Furniture        $211.70   ...   

   Customer_Satisfaction Engagement_with_Ads  Device_Used_for_Shopping  \
0                      7                 NaN                    Tablet   
1                      5                High                    Tablet   
2                      7                 Low                Smartphone   
3                      1                 NaN                Smartphone   
4                     10                 NaN                Smartphone   

   Payment_Method  Time_of_Purchase Discount_Used  \
0     Credit Card          3/1/2024          True   
1          PayPal         4/16/2024          True   
2      Debit Card         3/15/2024          True   
3           Other         10/4/2024          True   
4      Debit Card         1/30/2024         False   

  Customer_Loyalty_Program_Member  Purchase_Intent  Shipping_Preference  \
0                           False       Need-based        No Preference   
1                           False      Wants-based             Standard   
2                            True        Impulsive        No Preference   
3                            True       Need-based              Express   
4                           False      Wants-based        No Preference   

  Time_to_Decision  
0                2  
1                6  
2                3  
3               10  
4                4  

[5 rows x 28 columns]

In [5]:
data.shape

(1000, 28)

# 2. Data Cleaning

In [11]:
data.describe(include='all')

Customer_ID          Age  Gender Income_Level Marital_Status  \
count          1000  1000.000000    1000         1000           1000   
unique         1000          NaN       8            2              4   
top     37-611-6911          NaN  Female         High        Widowed   
freq              1          NaN     452          515            260   
mean            NaN    34.304000     NaN          NaN            NaN   
std             NaN     9.353238     NaN          NaN            NaN   
min             NaN    18.000000     NaN          NaN            NaN   
25%             NaN    26.000000     NaN          NaN            NaN   
50%             NaN    34.500000     NaN          NaN            NaN   
75%             NaN    42.000000     NaN          NaN            NaN   
max             NaN    50.000000     NaN          NaN            NaN   

       Education_Level Occupation Location Purchase_Category Purchase_Amount  \
count             1000       1000     1000              1000            1000   
unique               3          2      969                24             989   
top         Bachelor's       High     Oslo       Electronics        $178.04    
freq               341        517        4                54               2   
mean               NaN        NaN      NaN               NaN             NaN   
std                NaN        NaN      NaN               NaN             NaN   
min                NaN        NaN      NaN               NaN             NaN   
25%                NaN        NaN      NaN               NaN             NaN   
50%                NaN        NaN      NaN               NaN             NaN   
75%                NaN        NaN      NaN               NaN             NaN   
max                NaN        NaN      NaN               NaN             NaN   

        ...  Customer_Satisfaction Engagement_with_Ads  \
count   ...            1000.000000                 744   
unique  ...                    NaN                   3   
top     ...                    NaN                High   
freq    ...                    NaN                 270   
mean    ...               5.399000                 NaN   
std     ...               2.868454                 NaN   
min     ...               1.000000                 NaN   
25%     ...               3.000000                 NaN   
50%     ...               5.000000                 NaN   
75%     ...               8.000000                 NaN   
max     ...              10.000000                 NaN   

        Device_Used_for_Shopping  Payment_Method  Time_of_Purchase  \
count                       1000            1000              1000   
unique                         3               5               344   
top                      Desktop          PayPal          3/3/2024   
freq                         350             219                 8   
mean                         NaN             NaN               NaN   
std                          NaN             NaN               NaN   
min                          NaN             NaN               NaN   
25%                          NaN             NaN               NaN   
50%                          NaN             NaN               NaN   
75%                          NaN             NaN               NaN   
max                          NaN             NaN               NaN   

       Discount_Used Customer_Loyalty_Program_Member  Purchase_Intent  \
count           1000                            1000             1000   
unique             2                               2                4   
top             True                           False       Need-based   
freq             521                             509              256   
mean             NaN                             NaN              NaN   
std              NaN                             NaN              NaN   
min              NaN                             NaN              NaN   
25%              NaN                             NaN        

Missing Values

In [9]:
missing_values(data)

Missing Values  Percentage Missing  \
Customer_ID                                         0                 0.0   
Age                                                 0                 0.0   
Gender                                              0                 0.0   
Income_Level                                        0                 0.0   
Marital_Status                                      0                 0.0   
Education_Level                                     0                 0.0   
Occupation                                          0                 0.0   
Location                                            0                 0.0   
Purchase_Category                                   0                 0.0   
Purchase_Amount                                     0                 0.0   
Frequency_of_Purchase                               0                 0.0   
Purchase_Channel                                    0                 0.0   
Brand_Loyalty                                       0                 0.0   
Product_Rating                                      0                 0.0   
Time_Spent_on_Product_Research(hours)               0                 0.0   
Social_Media_Influence                            247                24.7   
Discount_Sensitivity                                0                 0.0   
Return_Rate                                         0                 0.0   
Customer_Satisfaction                               0                 0.0   
Engagement_with_Ads                               256                25.6   
Device_Used_for_Shopping                            0                 0.0   
Payment_Method                                      0                 0.0   
Time_of_Purchase                                    0                 0.0   
Discount_Used                                       0                 0.0   
Customer_Loyalty_Program_Member                     0                 0.0   
Purchase_Intent                                     0                 0.0   
Shipping_Preference                                 0                 0.0   
Time_to_Decision                                    0                 0.0   

                                      Datatype  
Customer_ID                             object  
Age                                      int64  
Gender                                  object  
Income_Level                            object  
Marital_Status                          object  
Education_Level                         object  
Occupation                              object  
Location                                object  
Purchase_Category                       object  
Purchase_Amount                         object  
Frequency_of_Purchase                    int64  
Purchase_Channel                        object  
Brand_Loyalty                            int64  
Product_Rating                           int64  
Time_Spent_on_Product_Research(hours)  float64  
Social_Media_Influence                  object  
Discount_Sensitivity                    object  
Return_Rate                              int64  
Customer_Satisfaction                    int64  
Engagement_with_Ads                     object  
Device_Used_for_Shopping                object  
Payment_Method                          object  
Time_of_Purchase                        object  
Discount_Used                             bool  
Customer_Loyalty_Program_Member           bool  
Purchase_Intent                         object  
Shipping_Preference                     object  
Time_to_Decision                         int64

At this point there are several things worth noting down:
1. Only 2 features with missing values - Social_Media_Influence (24.7%),
Engagement_with_Ads (25.6%)
2. 8 unique values for gender, female is the most frequent answer
3. Most locations are unique - map it out, or change to country
4. Purchase amount needs to be converted to numerical
5. Time of purchase needs to be separate into more feature

2. age ranges between 18 - 50, with median being 34
4. 2 values for income, high being the most frequent
5. 4 values for martial state, widowed is the most frequent
6. 3 education levels, bachelor's is the most frequest
7. 2 occupations
9. 24 purchase categories
11. Frequency of purchase ranges between 2-12
12. 3 purchase channels
13. Brand loyalty between 1-5, mean being 3
14. Product rating between 1-5, mean being 3
15. Time spent researching product 0-2 (hours)
16. 3 values for social media influence, high being the most frequent
17. 3 values for Discount sensitivity, very sensitive the most frequent
18. Return rate 0-2
19. Customer satisfaction 0-10, 5 is median
20. Engagement with Ads 3 values, high is frequent
21. Device used for shopping 3 values,
22. 5 ways of Payment method

1. pattern with missing values? same users
2. clean up genders
3. 